File: scripts/sequencer/components/SequencerEvalExpression.js
#pragma strict
#pragma downcast

public function evalBool( expression:String):boolean
{
	var result:boolean = eval( expression);
	gameObject.SendMessage( "setEvalResultBool", result);

	Debug.Log( "Evaluated Expression: " + expression + " with result: " + result );
}

public function evalInt ( expression:String):int
{
	var result:int = eval( expression);
	gameObject.SendMessage( "setEvalResultInt", result);
	
	Debug.Log( "Evaluated Expression: " + expression + " with result: " + result );
}

public function evalFloat( expression:String):float
{
	var result:float = eval( expression);
	gameObject.SendMessage( "setEvalResultFloat", result);
	
	Debug.Log( "Evaluated Expression: " + expression + " with result: " + result );
}

File: Assets/increscent/sequencer/scripts/components/SequencerEvalExpression.cs
using TinyExe;
using UnityEngine;

public class SequencerEvalExpression : MonoBehaviour
{
    Expression exp;

    private SequencePlayer player;
    private void Awake()
    {
        player = GetComponent<SequencePlayer>();
    }

    public void evalBool(string expression)
    {
        exp = new Expression(expression);
        var result = exp.Eval();
        string string_result = result.ToString();
        bool bool_result;
        bool_result = string_result == "1";
        if (bool_result == false)
        {
            bool_result = string_result == "True";
        }

        player.setEvalResultBool(bool_result);
        
        Debug.Log("Evaluated Expression: " + expression + " with result: " + result);
    }
    
    public void evalInt(string expression)
    {
        exp = new Expression(expression);
        var result = exp.Eval();

        player.setEvalResultInt(int.Parse(result.ToString()));
        
        Debug.Log("Evaluated Expression: " + expression + " with result: " + result);
    }
    
    public void evalFloat(string expression)
    {
        exp = new Expression(expression);
        var result = exp.Eval();

        player.setEvalResultFloat( float.Parse(result.ToString()));
        
        Debug.Log("Evaluated Expression: " + expression + " with result: " + result);
    }
}
